src/Strategy/Node/Pnpm/PnpmLock.hs

```diff
@@ -123,21 +130,47 @@ data PnpmLockfile = PnpmLockfile
   { importers :: Map Text ProjectMap
   , packages :: Map Text PackageData
   , lockFileVersion :: PnpmLockFileVersion
+  , lockFileSnapshots :: PnpmLockFileSnapshots
+  -- ^ Dependency graph in lockfile version > 9
   }
   deriving (Show, Eq, Ord)
 
+type SnapshotDepName = Text
+
+type SnapShotDepRev = Text
+
+-- | Lockfile versions > 9 use snapshots to represent the dependency graph.
```

Comment:
Can you add an example of what this looks like? Just a snippet of the lockfile so that we can see the structure